blockprod/src/detail/mod.rs

```diff
@@ -242,6 +262,18 @@ impl BlockProduction {
         Ok(best_block_index)
     }
 
+    pub fn generate_custom_id(&self, input_data: &GenerateBlockInputData) -> CustomId {
+        match input_data {
+            GenerateBlockInputData::PoS(pos_input_data) => {
+                pos_input_data.stake_public_key().encode()
```

Comment:
not sure how important that is but it's confusing that PoS returns the same value every time. Given that struct is called "generate" and for PoW it actually generates random value. Is it intended?

Reply:
Yeah. So the function "generates" for PoW, but for PoS the value is fixed. The struct has a "Generate" prefix because it's the struct for the "generate_block()" RPC call.